arctic/arctic.py

```diff
@@ -202,6 +202,34 @@ def list_libraries(self):
                         libs.append(coll[:-1 * len(self.METADATA_COLL) - 1])
         return libs
 
+    def library_exists(self, library):
+        """
+        Check whether a given library exists.
+
+        Parameters
+        ----------
+        library : `str`
+            The name of the library. e.g. 'library' or 'user.library'
+
+        Returns
+        -------
+        `bool`
+            True if the library with the given name already exists, False otherwise
+        """
+        exists = False
+        try:
+            # This forces auth errors, and to fall back to the slower "list_collections"
+            ArcticLibraryBinding(self, library).get_library_type()
+            # This will obtain the library, if no exception thrown we have verified its existence
+            self.get_library(library)
+            exists = True
+        except OperationFailure:
+            exists = library in self.list_libraries()
```

Comment:
This will only execute if the user doesn't have auth/creds installed for the library, even read-only.
Users usually check using list_libraries and then at a later point use the library.
I'd expect to not be the usual case.

Reply:
Ship it